File: backend/src/main/java/backend/mulkkam/intake/dto/IntakeHistoryResponse.java
package backend.mulkkam.intake.dto;

import backend.mulkkam.intake.domain.IntakeHistory;
import java.time.LocalDateTime;

public record IntakeHistoryResponse(
        Long id,
        LocalDateTime dateTime,
        int intakeAmount
) {
    public IntakeHistoryResponse(IntakeHistory intakeHistory) {
        this(
                intakeHistory.getId(),
                intakeHistory.getDateTime(),
                intakeHistory.getIntakeAmount().value()
        );
    }
}

File: android/app/src/main/java/com/mulkkam/data/remote/model/response/intake/IntakeHistoryResponse.kt
package com.mulkkam.data.remote.model.response.intake

import com.mulkkam.domain.model.intake.IntakeHistory
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalTime
import java.time.format.DateTimeFormatter

@Serializable
data class IntakeHistoryResponse(
    @SerialName("id")
    val id: Int,
    @SerialName("time")
    val dateTime: String,
    @SerialName("intakeAmount")
    val intakeAmount: Int,
)

fun IntakeHistoryResponse.toDomain() =
    IntakeHistory(
        id = id,
        dateTime = LocalTime.parse(this.dateTime, DateTimeFormatter.ofPattern("HH:mm:ss")),
        intakeAmount = intakeAmount,
    )
